examples_test.go

```diff
@@ -0,0 +1,288 @@
+package eris_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/rotisserie/eris"
+)
+
+// todo: manually test how stack corrections work when using a "global" error value
+//       that wasn't created via eris.NewGlobal (e.g. the root error stack will be
+//       wrong and might cause issues when trying to insert wrap frames into it)
+
+// Demonstrates JSON formatting of wrapped errors that originate from
+// external (non-eris) error types.
+func ExampleUnpackedError_ToJSON_external() {
+	// example func that returns an IO error
+	readFile := func(fname string) error {
```

Comment:
refactor to standard `func` definition like `func readFile(fname string) error` because runtime considers these functions as anonymous

Reply:
this prevents godoc from adding the funcs to the generated examples